private/pkg/app/appcmd/appcmd.go

```diff
@@ -167,36 +176,33 @@ func run(
 	// commands.
 	cobraCommand.CompletionOptions.DisableDefaultCmd = true
 
-	// If the root command is not the only command, add hidden bash-completion,
-	// fish-completion, zsh-completion, and manpages commands.
+	// If the root command is not the only command, add a hidden manpages command
+	// and a visible completion command.
 	if len(command.SubCommands) > 0 {
 		cobraCommand.AddCommand(
 			&cobra.Command{
-				Use:    "bash-completion",
-				Args:   cobra.NoArgs,
-				Hidden: true,
-				Run: func(*cobra.Command, []string) {
-					runErr = cobraCommand.GenBashCompletion(container.Stdout())
-				},
-			},
-		)
-		cobraCommand.AddCommand(
-			&cobra.Command{
-				Use:    "fish-completion",
-				Args:   cobra.NoArgs,
-				Hidden: true,
-				Run: func(*cobra.Command, []string) {
-					runErr = cobraCommand.GenFishCompletion(container.Stdout(), true)
-				},
-			},
-		)
-		cobraCommand.AddCommand(
-			&cobra.Command{
-				Use:    "zsh-completion",
-				Args:   cobra.NoArgs,
-				Hidden: true,
-				Run: func(*cobra.Command, []string) {
-					runErr = cobraCommand.GenZshCompletion(container.Stdout())
+				Use:   "completion <" + strings.Join(supportedShells, "|") + ">",
```

Comment:
These should probably just be nested commands, which eliminates the need for this variable, protects against case-matching typos/bugs being introduced, and even provides auto-completion support for this command.

Alternatively, can we not just delete line 177 instead of a custom completion command? It seems equivalent to what we're doing. If for some reason we don't want the default `completion` command, defining our own `completion` command is equivalent to setting `DisableDefaultCmd`: https://github.com/spf13/cobra/blob/e04ec725508c760e70263b031e5697c232d5c3fa/completions.go#L599-L608